### ISSUE TITLE
Don't suggest documenting exceptions in unit test code

### ISSUE THREAD

**Issue by rsuter** [CSharpAnalyzers/ExceptionalReSharper#12]
I'd like to tell Exceptional to disable processing on any classes that have the `[NUnit.Framework.TestFixture]` attribute.

https://exceptional.codeplex.com/workitem/11006

**Comment by jongleur1983** [comment]
Stumbled over the same problem with XUnit, so probably it would be good to ignore classes and/or methods configurable by e.g.
- name pattern,
- existing attributes (like [NUnit.Framework.TestFixture], [XUnit.FactAttribute] etc,)
- project (to excluce test projects)

For test code the "ignore attributed methods" could be collected in presets ("ignore XUnit test code", "ignore NUnit test code", ...)

**Comment by Lechus** [comment]
2016-08 still issue not fixed?

**Comment by jongleur1983** [comment]
@Lechus feel free to work on it, or try to contribute in useful way at least, instead of complaining about time others do not spend enough in your eyes.

**Comment by rsuter** [comment]
I will accept PRs but I currently do not have the time to working on this project directly...

**Comment by jongleur1983** [comment]
@rsuter still struggling with project structure/debugging/different ReSharper versions, but quite interested in the project. Once I solve my issues with how to smoothly work on it, I think about getting a core contributor (if you like)

**Comment by rsuter** [comment]
I don't know if there is a good debugging story with the latest ReSharper, maybe @citizenmatt knows more?

**Comment by jongleur1983** [comment]
I asked @citizenmatt about it on Twitter already, which solved my issues up to a certain point (trying to debug Exceptional with ReSharper 2016.2), now I get it started and installed in a test instance, but it doesn't show up anywhere yet.
Before I used the quite outdated ReSharper API documentation that still referred to a  devenv command line switch to use the extension to debug.

**Comment by citizenmatt** [comment]
[This is the page](https://www.jetbrains.com/help/resharper/sdk/Extensions/Plugins/Debugging.html), that explains how to get debugging working. It's up to date, and has links to other sections such as creating the test instance and installing ReSharper there.

Once you've got a test instance set up with ReSharper installed, you can just install your extension from a local folder, and you're set up to debug. Either attach the debugger, or launch VS directly in the debugger, passing in the name of the test instance. You can also simply copy your .dll to the installation folder, too, to make it a bit easier to build, test, edit.

If nothing is showing up, you'll want to check the [troubleshooting page](https://www.jetbrains.com/help/resharper/sdk/Extensions/Troubleshooting.html).

**Comment by rsuter** [comment]
Thanks @citizenmatt, I really miss the /ReSharper.Plugin flag which made debugging extremely simple... Now you always have to build, install, attach debugger, etc... its too time consuming to develop a R# extension.

**Comment by jongleur1983** [comment]
still no success for me unfortunately. (sorry for hijacking this thread for now).
Resharper install log contains this exception:

```--- EXCEPTION #1/1 [LoggerException]
  Message = “The Master Assembly Binder does not know this assembly, and neither does the GAC. Reference from assembly ReSharper.Exceptional.2016.1 to assembly JetBrains.ReSharper.Feature.Services could not be resolved.”
  ExceptionPath = Root
  ClassName = JetBrains.Util.LoggerException
  HResult = COR_E_APPLICATION=80131600```

All I could find about that sounds unreasonable as Exceptional should have worked before, so it shouldn't be necessary to add or remove some dependencies.

**Comment by rsuter** [comment]
did you install it via nuget?
i think you have to define your own nuget source (i.e. directory path) and install the locally compiled nuget (see .bat scripts)...

**Comment by jongleur1983** [comment]
that's what I did. I may have done anything wrong when trying to create the 2016.2 package though. Starting from scratch again now stashing all changes

**Comment by rsuter** [comment]
You have to upgrade the wave version with nuget, only changing it in packages.config is not enough...

**Comment by rsuter** [comment]
@lotosbin has a working fork with v2016.2 you can merge it from there...

**Comment by citizenmatt** [comment]
Yeah, update the JetBrains.ReSharper.SDK package via nuget (just changing packages.config won't work). It'll upgrade everything else, including the Wave package. Then yes, update the Wave dependency to "[6.0]" in your .nuspec.

**Comment by Aratirn** [comment]
The possibility of turning the plugin off for certain projects (i.e. test assemblies) would be really interesting. Specially when testing for exceptions.
